crates/polars-core/src/datatypes/dtype.rs

```diff
@@ -387,7 +387,17 @@ impl Display for DataType {
             #[cfg(feature = "object")]
             DataType::Object(s, _) => s,
             #[cfg(feature = "dtype-categorical")]
-            DataType::Categorical(_, _) => "cat",
+            DataType::Categorical(rev_map, _) => {
+                if let Some(rev_map) = rev_map {
+                    if rev_map.is_enum() {
+                        "enum"
+                    } else {
+                        "cat"
+                    }
+                } else {
+                    "cat"
+                }
+            },
```

Comment:
Repeating the `"cat"` seems less than ideal - I think this is better written as:

```suggestion
            DataType::Categorical(rev_map, _) => match rev_map {
                Some(r) if r.is_enum() => "enum",
                _ => "cat",
            },
```

Reply:
Lol, I knew there would be a more compact way to express that - it felt like the formatter was calling me out 🤣